Add progress preview control for modules

gkaragia opened the pull request:
### Changes proposed in this Pull Request

* A new control has been added to the module block.
![Sep-16-2020 10-23-18](https://user-images.githubusercontent.com/53191348/93305146-ca8dbc00-f806-11ea-9a96-1f86844b8d02.gif)

### Testing instructions

* Try creating a new course and add a module block.
* While editing the module block, observe the new setting in the sidebard.
* Modify the setting and observe that the module status is changed.
* Observe the module status on the frontend. For now the frontend uses mock data as we will need to refactor that part when we connect to the actual data.

gkaragia commented on includes/blocks/class-sensei-course-outline-block.php:
```diff
@@ -185,6 +191,12 @@ private function get_module_block_html( $block ) {
 			<section class="wp-block-sensei-lms-course-outline-module">
 				<header class="wp-block-sensei-lms-course-outline-module__name">
 					<h2 class="wp-block-sensei-lms-course-outline__clean-heading">' . $block['title'] . '</h2>
+					<div
```
This is probably going to be heavily refactored when we connect to the actual state so I did a very simple soltion for now.

gkaragia commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Module Block - Add status preview](https://trello.com/c/bOqGrDv3/587-module-block-add-status-preview)

gkaragia commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Module Block - Add status preview](https://trello.com/c/bOqGrDv3/587-module-block-add-status-preview)

gkaragia commented on includes/blocks/class-sensei-course-outline-block.php:
```diff
@@ -185,6 +190,40 @@ private function get_module_block_html( $block ) {
+
+		if ( $module_progress < 100 ) {
+			$module_status   = __( 'COMPLETED', 'sensei-lms' );
+			$indicator_color = '#63a95f';
```
The color is inline to more easily support setting it with block attributes in the future. I could add a CSS class instead.

jom reviewed (commented):
The current design for this uses bold white text color, which might be a bit better contrast. The in-progress background with white might need to be contrast checked. It looks like we're also not including the icons, which is fine, but is a change from current design.

gkaragia commented:
Took the colors from the card 😄  Did a contrast check and both white and light gray seem to be ok. There is a small issue for green and black but still readable after checking with grayscale. That said I agree that the white background looks nicer so I switched it and moved the color to a class.

> It looks like we're also not including the icons, 
Which icons do you refer to? Is it the arrow next to the progress indicator? If yes I'll add these in my next card/PR.

jom commented:
I saw the background colors in the card, but not explicitly the font color. It is more the green/black that seems too close. 

It looks like the icons (checkmark and progress) from the current Sensei template didn't make it into the new design:

<img width="817" alt="Screen Shot 2020-09-17 at 4 47 08 pm" src="https://user-images.githubusercontent.com/68693/93575290-60555280-f991-11ea-9f0f-10097ff8a4a9.png">

jom commented on includes/blocks/class-sensei-course-outline-block.php:
```diff
@@ -191,6 +196,39 @@ private function get_module_block_html( $block ) {
+		}
+
+		if ( $module_progress < 100 ) {
+			$module_status   = __( 'COMPLETED', 'sensei-lms' );
```
I wonder if it'd be better to put these like `Completed`/`In Progress` and then use `text-transform: uppercase;` in CSS to get the effect.

gkaragia replied:
Good idea! a5be66aed915883560496029e5e30ec02361a976

gkaragia commented:
Hey @donnapep we did some changes to the background of the progress indicator to better much the figma designs. From the colors that were on the card we switched the light gray to white and made the green a bit lighter. Let me know if you think that the colors on the card are better.

jom commented on assets/blocks/course-outline/module-block/radio-group-control.js:
```diff
@@ -0,0 +1,41 @@
+ * @param {Function} props.setPreviewStatus A callback which is called when a new option is selected.
+ */
+export const RadioGroupControl = ( { previewStatus, setPreviewStatus } ) => {
+	const id = 'inspector-radio-group-control';
```
Should this be named based on what it is providing? `ModuleStatusControl`/`inspector-module-status-control` or something like that? We'll have a similar one for Lessons so it might be helpful to have the IDs and names diverge.

gkaragia replied:
I started to implemented this component as a reusable one but it proved to much work so I dropped the idea 😄  The names renamed though.

jom replied:
Yeah, I suppose the actual component is namespaced enough that for now it is fine.

jom reviewed (dismissed):
Looks good! Approving, but if @donnapep needs a change I can re-approve.

yscik commented on assets/blocks/course-outline/module-block/radio-group-control.js:
```diff
@@ -0,0 +1,41 @@
+import {
+	__experimentalRadio as Radio,
+	__experimentalRadioGroup as RadioGroup,
+	BaseControl,
```
Unfortunately these don't exist in Wordpress 5.3, resulting in an error when opening the block settings. 

An alternative may be just two buttons with toggling an active class, like `is-active: 'in-progress' === previewStatus`. — Which too is different in 5.3, named `is-pressed` vs `is-active` (but here it's safe to just set both)

gkaragia replied:
Switched to using ButtonGroup instead. Also took the chance to rename it to something more meaningfull and to switch to booleans.

jom reviewed (approved):
I like the button group change 👍
